week4/assignment/src/pages/SignUp.tsx

```diff
@@ -0,0 +1,130 @@
+import MainContainer from "../components/MainContainer";
+import { useState } from "react";
+import { useNavigate } from "react-router";
+import { getFirstPasswordError } from "../utils/getFirstPasswordError";
+import { getIsFull } from "../utils/getIsFull";
+import CreateId from "../components/signupSteps/CreateId";
+import CreatePasswd from "../components/signupSteps/CreatePasswd";
+import CreateNickname from "../components/signupSteps/CreateNickname";
+import { SIGN_UP_STEP, type SignUpStep } from "../constants/signupStep";
+import { signup } from "../apis/auth";
+
+const SignUp = () => {
+  const nav = useNavigate();
+  const [id, setId] = useState("");
+  const [passWd, setPassWd] = useState("");
+  const [confirmPassWd, setConfirmPassWd] = useState("");
+  const [nickName, setNickName] = useState("");
+  const [step, setStep] = useState<SignUpStep>(SIGN_UP_STEP.CREATE_ID);
+
+  const handleSignup = async () => {
+    try {
+      await signup({
+        loginId: id,
+        password: passWd,
+        nickname: nickName,
+      });
+      alert(`${nickName}님 반갑습니다! 회원가입에 성공했어요.`);
+      nav("/login");
+    } catch (error) {
+      alert("회원가입 실패.");
+    }
+  };
+
+  const handleIdChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const value = e.target.value;
+    setId(value);
+  };
+
+  const handlePassWdChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const value = e.target.value;
+    setPassWd(value);
+  };
+
+  const handleConfirmPassWdChange = (
+    e: React.ChangeEvent<HTMLInputElement>
+  ) => {
+    const value = e.target.value;
+    setConfirmPassWd(value);
+  };
+
+  const nextStep = () => {
+    switch (step) {
+      case SIGN_UP_STEP.CREATE_ID:
+        setStep(SIGN_UP_STEP.CREATE_PASSWD);
+        break;
+      case SIGN_UP_STEP.CREATE_PASSWD:
+        setStep(SIGN_UP_STEP.CREATE_NICKNAME);
+        break;
+    }
+  };
+
+  const onKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    if (e.key === "Enter" && isFull) {
+      if (
+        step === SIGN_UP_STEP.CREATE_ID ||
+        step === SIGN_UP_STEP.CREATE_PASSWD
+      )
+        nextStep();
+      else if (step === SIGN_UP_STEP.CREATE_NICKNAME) handleSignup();
+    }
+  };
```

Comment:
```suggestion
const onKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
  if (e.key !== "Enter" || !isFull) {
    return; 
  }

  if (step === SIGN_UP_STEP.CREATE_NICKNAME) {
    handleSignup();
    return;
  }

  if (step === SIGN_UP_STEP.CREATE_ID || step === SIGN_UP_STEP.CREATE_PASSWD) {
    nextStep();
  }
};
```

중첩 if문이 가독성을 떨어뜨리고 있어요,
early return으로 개선할 수 있어 보여요!